### PR TITLE
Add support for multivector derivative operators

### PR DESCRIPTION
**Pull request by eric-wieser**
This is what all my `Dop` work in the past was building up to.

### CODE REVIEW

File: galgebra/ga.py [hunk 1]
```diff
@@ -496,24 +496,53 @@ def coord_vec(self) -> Expr:
...
+            r_blade /= (self.e_sq**len(index))
+
+        return r_blade
+
+    @_cached_property
+    def _reciprocal_blade_dict(self) -> lazy_dict:
+        """ A dictionary mapping basis blades to their reciprocal blades. """
+        if self.r_basis is None:
+            self._build_reciprocal_basis(self.gsym)
+        return lazy_dict({}, self._reciprocal_of_basis_blade)
```
Comment (eric-wieser) [inline galgebra/ga.py:525]: I've left these private for now because I don't want to commit to the names.

File: galgebra/ga.py [hunk 2]
```diff
@@ -496,24 +496,50 @@ def coord_vec(self) -> Expr:
...
+        return r_blade
+
+    @_cached_property
+    def _reciprocal_blade_dict(self) -> lazy_dict:
+        """ A dictionary mapping basis blades to their reciprocal blades. """
         if self.r_basis is None:
             self._build_reciprocal_basis(self.gsym)
+        return lazy_dict({}, self._reciprocal_of_basis_blade)
+
+    def make_grad(self, a, cmpflg=False) -> mv.Dop:  # make gradient operator with respect to vector a
```
Comment (utensil) [inline galgebra/ga.py:527]: The comment is still saying "vector a"

Comment (utensil) [inline galgebra/ga.py:527]: The comment is still saying "vector a"

**Comment by eric-wieser** [comment]
Will write them next week - thought I'd just get the changes I wrote on friday up for review.

File: galgebra/ga.py [hunk 3]
```diff
@@ -496,24 +496,49 @@ def coord_vec(self) -> Expr:
...
+        Compute the reciprocal :math:`e^I` of a basis blade :math:`e_I`.
+
+        This is a blade :math:`e^I` such that :math:`\left<\e^Ie_J\right> = \delta_I^J`
+        (:cite:`Hestenes`, eq 3.19).
+        """
+        index = self.indexes_to_blades_dict.inverse[blade]
+        r_blade = reduce(self.wedge, [
+            self.r_basis[i] for i in index[::-1]
+        ], S.One)
+        r_blade = r_blade.simplify()
```
Comment (utensil) [inline galgebra/ga.py:510]: I wanted to reduce `simplify()` calls to a minimal, is there any reason to call simplify here (or could there be a cache?)

Comment (eric-wieser) [reply galgebra/ga.py:510]: This function is cached - which is why I call simplify here.

Comment (utensil) [reply galgebra/ga.py:510]: Oh

**Comment by eric-wieser** [comment]
Will add some tests on Monday anyway.

**Pull request merged**